src/compose/app.ts

```diff
@@ -753,19 +752,15 @@ export class App {
 				modules: await pathExistsOnHost('/lib/modules'),
 			}))(),
 			(async () =>
-				_.trim(
-					await fs.readFile(
-						path.join(constants.rootMountPoint, '/etc/hostname'),
-						'utf8',
-					),
-				))(),
+				(await config.get('hostname')) ??
+				(await (await fs.readFile('/etc/hostname', 'utf-8')).trim()))(),
```

Comment:
I can't make code suggestions on code you didn't modify so I have to past the whole thing... does the following work ? my compiler isn't complaining:

```
		const [
			opts,
			supervisorApiHost,
			hostPathExists,
			hostnameOnHost,
		] = await Promise.all([
			config.get('extendedEnvOptions'),
			dockerUtils
				.getNetworkGateway(constants.supervisorNetworkInterface)
				.catch(() => '127.0.0.1'),
			(async () => ({
				firmware: await pathExistsOnHost('/lib/firmware'),
				modules: await pathExistsOnHost('/lib/modules'),
			}))(),
      (await config.get('hostname') ?? await fs.readFile('/etc/hostname', 'utf-8')).trim(),
		]);
```

Reply:
Looks better, I'll change it